src/server/game/Movement/MovementGenerators/ConfusedMovementGenerator.cpp

```diff
@@ -26,8 +26,12 @@
 #include "Random.h"
 
 template<class T>
-ConfusedMovementGenerator<T>::~ConfusedMovementGenerator()
+ConfusedMovementGenerator<T>::ConfusedMovementGenerator() : _timer(0), _x(0.f), _y(0.f), _z(0.f)
 {
+    MovementGenerator::Mode = MOTION_MODE_DEFAULT;
```

Comment:
pretty sure you are supposed to use `this->` in this situation

Reply:
that sounds right, it did never make too much sense to mee either, like it works, but thats how u declare a local variable? or call a static one? dunnow, will change 👍

Reply:
Because this (or base) class is a template its "dependent name lookup", by qualifying it with `this->` you change when compiler resolves it (template two phase stuff)

Reply:
got it 👍